File: src/main/java/com/cartrecipes/controller/dto/recipe/DetailedRecipeDto.java
package com.cartrecipes.controller.dto.recipe;

import com.cartrecipes.controller.dto.product.ProductDto;

import java.util.List;

public record DetailedRecipeDto(Long id,
                                String name,
                                Integer priceInCents,
                                List<ProductDto> products)  {
}

File: src/main/kotlin/com/cartrecipes/controller/dto/recipe/DetailedRecipeDto.kt
package com.cartrecipes.controller.dto.recipe

import com.cartrecipes.controller.dto.product.ProductDto

class DetailedRecipeDto {
    var id: Long = 0
    var name: String? = null
    var priceInCents: Int = 0
    var products: List<ProductDto> = emptyList()
}